src/features/editable-table/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { localStorageController } from '@/shared/utils';
+import { VariablesData } from '@/types/types';
+import { Button, Empty, Flex, Form, Popconfirm, Space, Table, TableProps, Typography } from 'antd';
+import { useEffect, useState } from 'react';
+import EditableCell from './editable-cell';
+import TableOperation from './operation';
+import { useTranslations } from 'next-intl';
+const { Text } = Typography;
+
+export default function EditableTable() {
+  const t = useTranslations('Variables');
+  const [form] = Form.useForm();
+  const [data, setData] = useState<VariablesData[]>([]);
+  const [isAddItem, setIsAddItem] = useState(false);
+  const [editingKey, setEditingKey] = useState<number>(0);
+
+  const editItem = async (record: Partial<VariablesData> & { key: number }) => {
+    form.setFieldsValue(record);
+    setEditingKey(record.key);
+  };
+
+  const saveItem = async (key: React.Key) => {
+    const value = (await form.validateFields()) as VariablesData;
+    setData((prev) => prev.map((item) => (item.key === key ? { ...item, ...value } : item)));
+    setEditingKey(0);
+    setIsAddItem(false);
+  };
```

Comment:
```suggestion
  const saveItem = async (key: React.Key) => {
    try {
      const value = (await form.validateFields()) as VariablesData;
      setData((prev) => prev.map((item) => (item.key === key ? { ...item, ...value } : item)));
      setEditingKey(0);
      setIsAddItem(false);
    } catch {
    }
  };
```

<img width="1181" height="583" alt="image" src="https://github.com/user-attachments/assets/c66cba03-ba53-490e-9dc0-79f91b562508" />

This is the error that I get each time I try to save a variable with one of the fields being empty. The error does not happen if I use try/catch inside of that function (even if catch does nothing). Another solution would be to disable the save button until one of the fields has some data.

Reply:
I knew about this method, but I was so confused by this empty catch.
To be honest, I couldn't come up with a better solution, so I did as you suggested.